--- a/includes/ghc.mk
+++ b/includes/ghc.mk
@@ -177,7 +177,8 @@
 
 INSTALL_LIBS += $(includes_GHCCONSTANTS_HASKELL_VALUE)
 
-DERIVE_CONSTANTS_FLAGS += --gcc-program "$(CC)"
+DERIVE_CONSTANTS_FLAGS += --gcc-program $(firstword $(CC))
+DERIVE_CONSTANTS_FLAGS += $(addprefix --gcc-flag$(space),$(wordlist 2, $(words $(CC)) ,$(CC)))
 DERIVE_CONSTANTS_FLAGS += $(addprefix --gcc-flag$(space),$(includes_CC_OPTS) -fcommon)
 DERIVE_CONSTANTS_FLAGS += --nm-program "$(NM)"
 ifneq "$(OBJDUMP)" ""
